src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Helpers.cs

```diff
@@ -387,5 +394,58 @@ internal static int HashLength(HashAlgorithmName hashAlgorithmName)
                 throw new ArgumentOutOfRangeException(nameof(hashAlgorithmName));
             }
         }
+
+        internal static PbeParameters MapExportParametersToPbeParameters(Pkcs12ExportPbeParameters exportParameters)
+        {
+            return exportParameters switch
+            {
+                Pkcs12ExportPbeParameters.Pkcs12TripleDesSha1 => Windows3desPbe,
+                Pkcs12ExportPbeParameters.Default or Pkcs12ExportPbeParameters.Pbes2Aes256Sha256 => WindowsAesPbe,
+                _ => throw new CryptographicException(),
+            };
+        }
+
+        internal static void VerifyExportParameters(Pkcs12ExportPbeParameters exportParameters)
+        {
+            if (exportParameters is < Pkcs12ExportPbeParameters.Default or > Pkcs12ExportPbeParameters.Pbes2Aes256Sha256)
+            {
+                throw new ArgumentOutOfRangeException(nameof(exportParameters));
+            }
+        }
+
+        internal static void VerifyExportParameters(PbeParameters exportParameters)
```

Comment:
Is the name of this method over-reaching?  The SHA-3 limitation comment says it's just for PKCS12, so it doesn't seem like this is shared with AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey.  Maybe wants to be `VerifyPkcs12ExportParameters` (then rename its sister method for cleanliness)?

Reply:
Probably. Originally I added this method on `X509Certificate` where I think the name was more appropriately understood, but ended up moving it to `Helpers` when I got around to X509Cert2Collection, but didn't pause to think of the name.